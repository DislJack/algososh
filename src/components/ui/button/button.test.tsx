import { Button } from "./button";
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';

describe('Тестирование рендеринга кнопки: а) с текстом, б) без текста, в) блокирование, г) с индикатором загрузки', () => {
  it('Тестирование рендеринга кнопки с текстом', () => {
    const button = renderer
    .create(<Button text="текст кнопки" />)
    .toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Тестирование рендеринга кнопки без текста', () => {
    const button = renderer
    .create(<Button />)
    .toJSON()
    expect(button).toMatchSnapshot();
  })

  it('Тестирование рендеринга заблокированной кнопки', () => {
    const button = renderer
    .create(<Button disabled={true} />)
    .toJSON()
    expect(button).toMatchSnapshot();
  })

  it('Тестирование рендеринга кнопки с загрузкой', () => {
    const button = renderer
    .create(<Button isLoader={true} />)
    .toJSON()
    expect(button).toMatchSnapshot();
  });

  it('Тестирование клика по кнопке', () => {
    const animationFunction = jest.fn();
    // Реализация клика должна быть сделана не так как тестирование выше
    render(<Button text="Текст" onClick={animationFunction} />);
    const button = screen.getByText('Текст');
    fireEvent.click(button);
    expect(animationFunction).toBeCalled();
  })
})