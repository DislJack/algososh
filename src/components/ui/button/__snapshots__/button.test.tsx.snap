// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование рендеринга кнопки: а) с текстом, б) без текста, в) блокирование, г) с индикатором загрузки Тестирование рендеринга заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование рендеринга кнопки: а) с текстом, б) без текста, в) блокирование, г) с индикатором загрузки Тестирование рендеринга кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование рендеринга кнопки: а) с текстом, б) без текста, в) блокирование, г) с индикатором загрузки Тестирование рендеринга кнопки с загрузкой 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование рендеринга кнопки: а) с текстом, б) без текста, в) блокирование, г) с индикатором загрузки Тестирование рендеринга кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    текст кнопки
  </p>
</button>
`;
